### PR TITLE
Node-ify lifetimes

### PR DESCRIPTION
**Pull request by Veykril**
Let's see if this passes the tests 🤞 

Depends on https://github.com/rust-analyzer/ungrammar/pull/15

### CODE REVIEW

**Comment by matklad** [comment]
bors d+

here as well

@Veykril feel free to merge this and the correspoinding ungrammar PR as soon as you'll have time!

**Comment by Veykril** [comment]
Rebased on top of #6897

File: Cargo.toml [hunk 1]
```diff
@@ -16,7 +16,7 @@ text-size.opt-level = 3
 miniz_oxide.opt-level = 3
 
 [profile.release]
-incremental = true
+incremental = true
```
Comment (matklad) [inline Cargo.toml:19]: trailing ws?

Comment (Veykril) [reply Cargo.toml:19]: Oh, for some reason whatever formats that in vscode for me adds that in and I'm not sure why 😅

Comment (matklad) [reply Cargo.toml]: One of these days, I'll make a proper lossless toml library...

Comment (Veykril) [reply Cargo.toml]: There actually is a lossless toml library in rust(it even uses rowan) https://github.com/tamasfe/taplo and I just noticed I use the vscode extension of that but it seems the formatter for that doesn't play nicely yet.

Comment (lnicola) [reply Cargo.toml]: Strange, Even Better TOML works fine for me (with some consistent config tweaking).

**Pull request merged**